Model/Reclaim.php

```diff
@@ -127,6 +136,81 @@ public function productinspector($start_id, $end_id){
         return $response;
 
     }
+
+    public function getSubscribersById($start_id, $end_id, $storeId=null)
+    {
+        if (!$start_id || !$end_id ){ 
+            throw new NotFoundException(__('Please provide start_id and end_id'));
+        }
+
+        if ($start_id > $end_id){
+            throw new NotFoundException(__('end_id should be larger than start_id'));
+        }
+
+        $storeIdFilter = $this->_storeFilter($storeId);
+
+        $subscriberCollection =$this->subscriberCollection->create()
+            ->addFieldToFilter('subscriber_id', ['gteq' => (int)$start_id])
+            ->addFieldToFilter('subscriber_id', ['lteq' => (int)$end_id])
+            ->addFieldToFilter('store_id', [$storeIdFilter => $storeId]);
+
+        $response = $this->_packageSubscribers($subscriberCollection);
+
+        return $response;
+    }
+
+    public function getSubscribersByDateRange($start, $until, $storeId=null)
+    {
+        
+        if (!$start || !$until ){ 
+            throw new NotFoundException(__('Please provide start and until param'));
+        }
+        // start and until date formats
+        // $until = '2019-04-25 18:00:00';
+        // $start = '2019-04-25 00:00:00';
+
+        // don't want any big queries, we limit to 10 days
+        $until_date = strtotime($until);
+        $start_date = strtotime($start);
+        $datediff = $until_date - $start_date;
+
+        if (abs(round($datediff / (60 * 60 * 24))) > self::MAX_QUERY_DAYS){
+            throw new NotFoundException(__('Please don\'t query for more than 10 days'));
+        }
```

Comment:
maybe change the wording on this to something like "Cannot query more than 10 days".